mantidproject/mantid
# Eng Fitting Tab - Plot Selected Workspaces from Table

**Description of work.**
Adds a plot area to the fitting tab of the Engineering Diffraction 2 GUI.
Ticking the check boxes in the table allows for the plotting of the associated workspaces. 

**To test:**
1. Add this line to "mantid.user.properties": `mantidqt.python_interfaces=Direct/DGS_Reduction.py Direct/DGSPlanner.py Direct/PyChop.py Direct/MSlice.py SANS/ORNL_SANS.py Utility/TofConverter.py Reflectometry/ISIS_Reflectometry_Old.py Diffraction/Engineering_Diffraction_2.py Diffraction/Powder_Diffraction_Reduction.py Utility/FilterEvents.py Diffraction/HFIR_4Circle_Reduction.py Utility/QECoverage.py SANS/ISIS_SANS.py Muon/Frequency_Domain_Analysis.py Muon/Elemental_Analysis.py Muon/Frequency_Domain_Analysis_Old.py Muon/Muon_Analysis.py` to allow access to the Engineering Diffraction 2 GUI.
1. Open the Engineering Diffraction 2 GUI (Interfaces -> Diffraction -> Engineering Diffraction 2)
1. Perform a calibration (unless you have one already, then load that one) using V:307521 S: 305738
2. Do some focusing (unless you already have files, then use them) for each file and bank so you have a range of files. S: 305761, 305738
3. Open the fitting tab 
4. Load a range of file names and types.
5. Check that all selected files were loaded.
6. Check that the table is populated with the information about the loaded files. 
7. Check that the remove buttons work in the way you would expect. 
8. Check that checking the boxes plots the values onto the integrated plot. 
9. Check that removing also removes from the plot if the workspace was plotted. 

Fixes #27779 .

*This does not require release notes* because **this GUI is not yet user facing**

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
